torchgen/gen_functionalization_type.py

```diff
@@ -395,14 +395,17 @@ def emit_view_functionalization_body(
           return {reverse_lambda.inner_call()}
         }}
       );
-      at::functionalization::impl::mutate_view_meta({view_tensor_name}, view_meta);
       {return_type} reference_tensor_output;
       {{
         at::AutoDispatchSkipFunctionalize guard;
         {meta_conversion_str}
         reference_tensor_output = at::_ops::{noop_api_name}::call({', '.join(meta_call_args)});
       }}
+      at::functionalization::impl::mutate_view_meta({view_tensor_name}, view_meta);
       // See  Note [Propagating strides in the functionalization pass]
+      // XLA/LTC don't implement the logic to propagate strides correctly, so we need to rely
+      // on a reference implementation here (instead of relying on the output from the forward lambda
+      // having the correct stride info)
```

Comment:
My first attempt was just to say "hey we don't actually need to run the meta tensor reference implementation! We can just rely on sizes/strides/storage_offset being correct after running the view op on the inner tensor!

We can't rely on that though because XLA/LTC's implementations for `transpose` won't actually properly compute strides for you.